src/react/src/actions/claimedFacilityDetails.js

```diff
@@ -97,8 +155,22 @@ export function submitClaimedFacilityDetailsUpdate(claimID) {
                     isInt(data.facility_female_workers_percentage)
                         ? data.facility_female_workers_percentage
                         : null,
+                parent_company_name: parentCompanyName,
+                opening_date: data.opening_date || null,
+                closing_date: data.closing_date || null,
+                estimated_annual_throughput:
+                    data.estimated_annual_throughput || null,
             },
         );
+
+        // Only include energy fields if the corresponding checkbox is enabled.
+        energyFields.forEach(fieldName => {
+            const enabledKey = `${fieldName}_enabled`;
+            if (data[enabledKey]) {
+                updateData[fieldName] = data[fieldName] || null;
+            }
+        });
```

Comment:
Can we use `map` or something like that here instead of mutating the `updatedData` within the loop?